sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -54,7 +99,110 @@ export class JsonRpcProvider extends Provider {
     public skipDataValidation: boolean = false
   ) {
     super();
+
     this.client = new JsonRpcClient(endpoint);
+
+    this.wsEndpoint = getWebsocketUrl(endpoint);
+
+    const socketOptions = { reconnect_interval: 3000, autoconnect: false };
+    this.wsClient = new WsRpcClient(this.wsEndpoint, socketOptions);
+    this.setupSocket();
```

Comment:
Can we move this call into the `connect` call?

Reply:
+1. This should not be called by default since most callers are not using event subscription